### ISSUE TITLE
Possible to have static files instead of dynamic script on the server?

### ISSUE THREAD

**Issue by gnotaras** [YahnisElsts/wp-update-server#29]
Hello,

This project is very interesting.

The only thing I am skeptical about is the fact that the server has to run a dynamic script and thus be prone to abuse.

At least, instead of the following GET arguments:

`
http://yourserver.com/wp-update-server/?action=get_metadata&slug=your-plugin
`

... would it be possible to have a permalink structure like:

`
http://yourserver.com/wp-update-server/get/metadata/your-plugin-slug/
`

.. and responses with proper `Cache-Control` headers, so as to be able to cache them on a CDN or a caching proxy?

I'd like to make clear that I am not aware of the Update API, so all of the above might be just nonsense.

To sum it up, is it possible to upload static files, which contain all the necessary data about the plugin status, on a server and still have `plugin update checker` check them and work as expected?

George

**Comment by YahnisElsts** [comment]
Yes, you can definitely use a static file with `plugin-update-checker`. In fact, that is exactly what I did myself before I wrote this update server. Just make a JSON file with [the required fields](https://github.com/YahnisElsts/plugin-update-checker#self-hosted-plugins) and point the update checker to that file. 

- [Example JSON file](http://1.shadowcdn.com/files/external-update-example/info.json)
- [Full list of supported fields](https://docs.google.com/spreadsheets/d/1eOBbW7Go2qEQXReOOCdidMTf_tDYRq4JfegcO1CBPIs/pub?authkey=CK7h9toK&hl=en&single=true&gid=0&output=html)

**Comment by gnotaras** [comment]
@YahnisElsts: I see. Thanks a lot for pointing me to these documents! Very helpful in order to understand the whole procedure.

**Comment by gnotaras** [comment]
Also, since the implementation is intentionally fully dynamic, please feel free to close this issue. Thanks for your reply.

**Comment by YahnisElsts** [comment]
All right, issue closed.
